File: ExpenseBook/src/net/nctucs/lazchi/marco79423/ExpenseBook/Globals.java
package net.nctucs.lazchi.marco79423.ExpenseBook;

class Globals
{
	public static final String LOG_TAG = "累死雞記帳";

	public static final String DATABASE_NAME = "accountbook.db";
	public static final int DATABASE_VERSION = 3;

	public static final String DROPBOX_KEY = "7hrz989zzhcyr9t";
	public static final String DROPBOX_SECRET = "zcyop6imgp7oq5i";

	public static final String DATE_FORMAT = "yyyy/MM/dd";

	public static class Expense
	{
		public static String ID = "_id";
		public static String PICTURE_BYTES = "picture_bytes";
		public static String SPEND_STRING = "spend_string";
		public static String DATE_STRING = "date_string";
		public static String CATEGORY = "category";
		public static String NOTE = "note";
	}

	public static class ExpenseTable
	{
		public static final String TABLE = "expenses";

		public static final String ID = Expense.ID;
		public static final String PICTURE_BYTES = Expense.PICTURE_BYTES;
		public static final String SPEND_STRING = Expense.SPEND_STRING;
		public static final String DATE_STRING = Expense.DATE_STRING;
		public static final String CATEGORY_ID = "category_id";
		public static final String NOTE = Expense.NOTE;

		public static final String CREATE_TABLE =
			"CREATE TABLE IF NOT EXISTS " + TABLE + "(" +
			ID + " INTEGER NOT NULL PRIMARY KEY," +
			PICTURE_BYTES + " BLOB," +
			SPEND_STRING + " TEXT," +
			DATE_STRING + " DATE," +
			CATEGORY_ID + " NUMERIC," +
			NOTE + " TEXT);";

		static final String DELETE_TABLE = "DROP TABLE IF EXISTS " + TABLE;
	}

	public static class CategoryTable
	{
		public static final String TABLE = "categories";

		public static final String ID = "_id";
		public static final String CATEGORY = Expense.CATEGORY;
		public static final String ORDER_ID = "order_id";

		public static final String CREATE_TABLE =
			"CREATE TABLE IF NOT EXISTS " + TABLE + "(" +
			ID + " INTEGER NOT NULL PRIMARY KEY," +
			CATEGORY + " TEXT," +
			ORDER_ID + " NUMERIC);";
		public static final String DELETE_TABLE = "DROP TABLE IF EXISTS " + TABLE;
	}

	public static class InfoTable
	{
		public static final String TABLE = "info";

		public static final String ID = "_id";
		public static final String APP_VERSION = "app_version";

		static final String CREATE_TABLE =
			"CREATE TABLE IF NOT EXISTS " + TABLE + "(" +
			ID + " _id INTEGER NOT NULL PRIMARY KEY," +
			APP_VERSION + " TEXT);";
		static final String DELETE_TABLE = "DROP TABLE IF EXISTS " + TABLE;
	}
}

File: ExpenseBook-pc/globals.cpp
#include "globals.h"

#include <QString>
#include <QSize>
#include <QColor>

namespace ExpenseBook
{
    const QString organization()
    {
        return "兩大類 x 兩大類 = 四大類";
    }

    const QString organizationDomain()
    {
        return "marco79423.lazchi.nctucs.net";
    }

    const QString application()
    {
        return "累死雞記帳";
    }

    const QString databasePathSetting()
    {
        return "databasePath";
    }

    const QString createExpenseTableString()
    {
        return "CREATE TABLE IF NOT EXISTS expenses( "
        "   _id INTEGER NOT NULL PRIMARY KEY, "
        "   picture_bytes BLOB, "
        "   spend NUMERIC, "
        "   data_string DATE, "
        "   category_id NUMERIC, "
        "   note TEXT"
        ");";
    }

    const QString createCategoryTableString()
    {
        return "CREATE TABLE IF NOT EXISTS categories( "
        "   _id INTEGER NOT NULL PRIMARY KEY, "
        "   category TEXT, "
        "   order_id NUMERIC "
                ");";
    }

    const char* pictureFormat()
    {
        return "JPG";
    }

    const QSize pictureSizeInExpenseListView()
    {
        return QSize(100, 60);
    }

    const QColor mainColor()
    {
        return QColor(56, 140, 168);
    }

    const QColor backgroundColor()
    {
        return QColor(200, 200, 200);
    }

    const QColor fontColor()
    {
        return QColor(85, 85, 85);
    }

    const QColor fontWhiteColor()
    {
        return QColor(255, 255, 255);
    }

    const QString dateFormat()
    {
        return "yyyy/MM/dd";
    }

    const QString insertExpenseString()
    {
        return "INSERT INTO expenses(picture_bytes, spend, date_string, category_id, note) VALUES (:picture_bytes, :spend, :date_string, :category_id, :note);";
    }

    const QString editExpenseString()
    {
        return "UPDATE expenses SET picture_bytes=:picture_bytes, spend=:spend, date_string=:date_string, category_id=:category_id, note=:note WHERE _id=:_id;";
    }

    const QString deleteExpenseString()
    {
        //_id 可能有 bug ，只好用 ? 代替
        return "DELETE FROM expenses WHERE _id=?;";
    }

}
